helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithmFactory.java

```diff
@@ -47,13 +62,17 @@ public static RebalanceAlgorithm getInstance(
     float evennessRatio = (float) evennessPreference / (evennessPreference + movementPreference);
     float movementRatio = (float) movementPreference / (evennessPreference + movementPreference);
 
-    Map<SoftConstraint, Float> softConstraints = ImmutableMap.<SoftConstraint, Float> builder()
-        .put(new PartitionMovementConstraint(), movementRatio)
-        .put(new InstancePartitionsCountConstraint(), 0.3f * evennessRatio)
-        .put(new ResourcePartitionAntiAffinityConstraint(), 0.1f * evennessRatio)
-        .put(new ResourceTopStateAntiAffinityConstraint(), 0.1f * evennessRatio)
-        .put(new MaxCapacityUsageInstanceConstraint(), 0.5f * evennessRatio).build();
+    List<SoftConstraint> l = ImmutableList.of(new PartitionMovementConstraint(),
+        new InstancePartitionsCountConstraint(), new ResourcePartitionAntiAffinityConstraint(),
+        new ResourceTopStateAntiAffinityConstraint(), new MaxCapacityUsageInstanceConstraint());
+    Map<SoftConstraint, Float> softConstraints = Maps.toMap(l, key -> {
+      String name = key.getClass().getSimpleName();
+      float weight = MODEL.get(name);
```

Comment:
Put default value here if MODEL does not contain the name. The default value should be 0 in my opinion.